File: src/main/java/com/totvs/template/Domain/Dto/Security/Users/UserDto.java
package com.totvs.template.Domain.Dto.Security.Users;

import com.totvs.template.Domain.Dto.Base.EntityBaseDto;
import com.totvs.template.Domain.Dto.Security.Roles.RoleDto;
import java.util.HashSet;
import java.util.Set;

public class UserDto extends EntityBaseDto {
	
	// Attributes
	public String username;
	public String password;
	public String token;
	public String mail;
	public String name;
	public String surname;
	public Set<RoleDto> roles = new HashSet<>();

	
	public String getToken() {
		return token;
	}
	public void setToken(String token) {
		this.token = token;
	}
	public String getMail() {
		return mail;
	}


	public void setMail(String mail) {
		this.mail = mail;
	}
	public String getName() {
		return name;
	}


	public void setName(String name) {
		this.name = name;
	}
	public String getPassword() {
		return password;
	}


	public void setPassword(String password) {
		this.password = password;
	}
	public Set<RoleDto> getRoles() {
		return roles;
	}


	public void setRoles(Set<RoleDto> roles) {
		this.roles = roles;
	}
	public String getSurname() {
		return surname;
	}


	public void setSurname(String surname) {
		this.surname = surname;
	}
	public String getUsername() {
		return username;
	}


	public void setUsername(String username) {
		this.username = username;
	}

}

File: src/client/src/shared/dtos/security/users/user-dto.ts
import { RoleDto } from "@shared/dtos/security/roles/role-dto";
import { SHA3 } from 'sha3';

export class UserDto {

	public id: number;
	public mail?: string;
	public name?: string;
	public password: string;
	public roles?: RoleDto[];
	public surname?: string;
	public username: string;

	public token: string;
	
	constructor(
		id?: number | null,
		username?: string | null,
		token?: string | null,
		roles?: RoleDto[] | null
	) {
		this.id = id;
		this.username = username;
		this.token = token;
		this.roles = roles;
	}

	static init(values: any) {
		var user : UserDto = new UserDto();
		return Object.assign(user, values)
	}

	static fromJson(json: Array<any> = []) {

		let items: Array<UserDto> = [];

		for (let values of json) {
			items.push(this.init(values));
		}

		return items;
	}

	static of(json: any = {}) {
		return this.init(json);
	}

	isAdmin(): boolean {
		if (!this.roles)
			return false;
		return this.roles.find(r => r.role === 'ADMIN') !== undefined;
	}

    /**
     * Check if logged user has one role
     */
	hasRole(role: string | Array<string>): boolean {
		if (!this.roles) return false;

		if (typeof role === 'string') {
			return (this.roles.find(r => r.role === role) !== undefined);
		} else if(typeof role === 'undefined') {
			return true;
		} else {
			const found = role.filter(rol => {
				return (this.roles.find(r => r.role == rol) !== undefined);
			});
			return found.length > 0;
		}
	}

	passwordToSHA3() : string{
		const hash = new SHA3(512);
        hash.update(this.password);
        return hash.digest('hex');
	}

	getUsername() {
		return this.username;
	}

	setUsername(username: string) {
		this.username = username;
	}

	getPassword() {
		return this.password;
	}

	setPassword(password: string) {
		this.password = password;
	}

	getId() {
		return this.id;
	}

	setId(name: string) {
		this.name = name;
	}

	getName() {
		return this.name;
	}

	setName(id: number) {
		this.id = id;
	}

	getToken() {
		return this.token;
	}

	setToken(token: string) {
		this.token = token;
	}

	setRoles(roles: RoleDto[]) {
		this.roles = roles;
	}

	getRoles() {
		return this.roles;
	}
}